core/telemetry/src/lib.rs

```diff
@@ -210,17 +232,24 @@ impl slog::Drain for TelemetryDrain {
 	type Err = ();
 
 	fn log(&self, record: &slog::Record, values: &slog::OwnedKVList) -> Result<Self::Ok, Self::Err> {
-		if let Some(inner) = self.inner.0.upgrade() {
-			let before = Instant::now();
-			let result = inner.worker.lock().log(record, values);
-			inner.polling_waker.wake();
-			if before.elapsed() > Duration::from_millis(50) {
-				warn!(target: "telemetry", "Writing a telemetry log took more than 50ms");
+		let before = Instant::now();
+
+		let serialized = slog_async::AsyncRecord::from(record, values);
+		// Note: interestingly, `try_send` requires a `&mut` because it modifies some internal value, while `clone()`
+		// is lock-free.
+		let result = match self.sender.clone().try_send(serialized) {
+			Ok(()) => Ok(()),
+			Err(err) => {
+				warn!(target: "telemetry", "Ignored telemetry message because of error on channel: {:?}", err);
+				Err(())
```

Comment:
I'm not sure we want to do this. `slog` will panic when it sees an error as we use `Drain::fuse`.

An error could be that the channel is full or closed.